bot/bot.py

```diff
@@ -407,11 +455,14 @@ def run_bot() -> None:
     application.add_handler(CallbackQueryHandler(set_chat_mode_handle, pattern="^set_chat_mode"))
 
     application.add_handler(CommandHandler("balance", show_balance_handle, filters=user_filter))
-    
-    application.add_error_handler(error_handle)
-    
+
+    if config.ignore_network_error:
+        application.add_error_handler(error_handle_except_network)
+    else:
+        application.add_error_handler(error_handle)
+
     # start the bot
-    application.run_polling()
+    application.run_polling(stop_signals=None)
```

Comment:
I think `stop_signals` is by default `None` [see in doc](https://docs.python-telegram-bot.org/en/stable/telegram.ext.application.html#telegram.ext.Application.run_polling)
So, this addition is not necessary.